dask_sql/physical/rex/core/literal.py

```diff
@@ -167,18 +167,17 @@ def convert(
             "TimestampNanosecond",
         }:
             unit_mapping = {
-                "Second": "s",
-                "Millisecond": "ms",
-                "Microsecond": "us",
-                "Nanosecond": "ns",
+                "TimestampSecond": "s",
+                "TimestampMillisecond": "ms",
+                "TimestampMicrosecond": "us",
+                "TimestampNanosecond": "ns",
             }
+            numpy_unit = unit_mapping.get(literal_type)
             literal_value, timezone = rex.getTimestampValue()
             if timezone and timezone != "UTC":
                 raise ValueError("Non UTC timezones not supported")
-            literal_type = SqlTypeName.TIMESTAMP
```

Comment:
after this `if` block, you can have:
```
elif timezone is None:
    literal_value = datetime.fromtimestamp(literal_value // 10**9)
```
(make sure to do a `from datetime import datetime`)

Reply:
btw, I chose 10**9 to convert to seconds, as suggested by https://stackoverflow.com/questions/45423917/pandas-errno-75-value-too-large-for-defined-data-type, but maybe milliseconds (or something else) would be generally better.

Reply:
Thanks @sarahyurick that fixes the regression locally for me. I have pushed this change.